### ISSUE TITLE
Check syntax show error 293 after save

### ISSUE THREAD

**Issue by cverbiest** [omecra/de.omecra.oedt.updatesite#66]
If I do a check syntax (Ctrl-Shift-C) immediately after a save (Ctrl-S) I frequently get an error 293 (see screenshot).

I guess there is a conflict between the compile triggered by the save and the check syntax.
If I repeat the check syntax it works fine.

I'm still using check syntax because the compile on save gives no visual feedback of success unless you're watching the console log

![image](https://user-images.githubusercontent.com/1312691/58960835-3e3dbe00-87a8-11e9-92b9-06c6ed3a90c8.png)

OEDT 9.0.3
OE 11.7.4

**Comment by lneumeier** [comment]
Does this happen with the progress editor too?

**Comment by cverbiest** [comment]
The progress editor gives
```
---------------------------
Check Syntax
---------------------------
The OpenEdge AVM is busy. Check Syntax cannot be performed at this time.
---------------------------
```

probably the same underlying issue but the message is clearer.

Would it be feasible to display the
![image](https://user-images.githubusercontent.com/1312691/59096748-dd36f700-891b-11e9-8513-7db327b19828.png)
popup in the editor if the current file compiled without invoking check syntax ?
This would give more feedback to the developers and remove the need for pressing check syntax immediately ?

**Comment by lneumeier** [comment]
I look into it, but the general concept is:
If there is an error, the editor get's an error marker. If there is no error - no marker.

**Comment by cverbiest** [comment]
I found that I don't need to save to get the error.
Check syntax without prior save can also show the same error 293.
I get it about 50% of the time if I use check syntax.

**Comment by Ton-Pe** [comment]
I have experienced the same problem. I tested this more and if I write some code and immediately after that check syntax using ctrl+shift+c), I can reproduce this almost every time.

**Comment by lneumeier** [comment]
Have you enabled "Check Syntax Automatically"?

**Comment by Ton-Pe** [comment]
Yes, I have.

**Comment by cverbiest** [comment]
"Check Syntax Automatically" is on,

**Comment by lneumeier** [comment]
Okay, "Check Syntax" copies the file to temp location and then get's compiled without the SAVE flag. After that, the file get's deleted. I assume, that there is some "racing conditions" involved, if two threads ("Auto Check Syntax" and "Check Syntax Action") is executed in parallel.

**Comment by lneumeier** [comment]
Fixed in 9.2.0.
